libr/core/cmd.c

```diff
@@ -2656,29 +2641,18 @@ static int r_core_cmd_subst_i(RCore *core, char *cmd, char *colon) {
 			r_core_block_size (core, tmpbsz);
 		}
 		if (is_arch_set) {
+			core->fixedarch = false;
```

Comment:
I'd save the original value at the beginning of the function and restore that here, instead of hardcoding it to false. Sometimes this function is recursively called, so you may end up with wrong `fixedarch` values if you hardcode the false value.